src/arrayMethodJoin.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
redundant, your str is empty string by default, so if there in no this.length, you will return empty string